aspnetcore/blazor/components/css-isolation.md

```diff
@@ -0,0 +1,160 @@
+
+`ChildExample.razor`:
+```razor
+@page "/child-example"
+
+<div>
+    <h1>Child Example</h1>
+    <MyChild />
+</div>
+```
+
+`MyChild.razor`:
+```razor
+@page "/first-child"
+
+<h1>A Child Component</h1>
+```
+
+Update the `h1` declaration in `MyComponent.razor.css` with the `::deep` combinator to signify the `h1` style declaration must apply to the parent component and its children.
+
+```css
+::deep h1 { 
+    color: red;
+}
+```
+
+The `h1` style now applies to the `ChildExample` and `FirstChild` components without the need to create separate scoped CSS files for the child components.
+
+> [!NOTE]
+> The `::deep` combinator works only with descendant elements. The following HTML structure applies the `h2` styles to components as expected:
+> 
+>```razor
+><div>
+>    <h1>Child Example</h1>
+>    <MyChild />
+></div>
+>```
+> In this scenario, ASP.NET Core applies the parent component's scope identifier to the `div` element, so it knows to inherit styles from the parent component.
+>
+>However, excluding the `div` element removes the descendant relationship, and the style will *not* be applied to the child component. 
+>
+>```razor
+><h1>Child Example</h1>
+><MyChild />
+>```
+
+## CSS preprocessor support
+
+CSS preprocessors are useful with improving CSS development by utilizing features like variables, nesting, modules, mixins, and inheritance. While CSS isolation does not *natively* support CSS preprocessors such as Sass or Less, integrating them is seamless as long as preprocessor compilation occurs before Blazor rewrites the CSS selectors during the build process.
+
+For example, in the Visual Studio Task Runner Explorer, configure existing preprocessor compilation as a **Before Build** task.
+
+Many third-party NuGet packages, such as [Delegate.SassBuilder](https://www.nuget.org/packages/Delegate.SassBuilder/), can compile SASS/SCSS files at the beginning of the build process before CSS isolation occurs—preventing any additional configuration.
+
+For example, when you add a `MyComponent.razor.scss` file, these tools can compile SASS changes to `MyComponent.razor.css` at the beginning of the build process. With that step complete, the isolation step is ready. 
+
+## CSS isolation configuration
+
+When working with isolated CSS, no configuration is required. However, configuration is available for advanced scenarios or when there are dependencies on other existing tools or workflows.
+
+### Customize scope identifier format
+
+By default, scope identifiers use the format `b-<10-character-string>`. To customize the scope identifier format, update the project file to a desired pattern.
+
+```xml
+<ItemGroup>
+    <None Update="MyComponent.razor.css" CssScope="my-custom-scope-identifier" />
+</ItemGroup>
+```
+
+In this example, the scope identifier changes from `b-<10-character-string>` to `my-custom-scope-identifier`.
```

Comment:
I would clarify that this is specific to CSS generated from `MyComponent.razor.css`.

Reply:
Changed to:

In this example, the CSS generated from `MyComponent.Razor.css` changes its scope identifier from `b-<10-character-string>` to `my-custom-scope-identifier`.